yt/geometry/oct_container.pyx

```diff
@@ -1073,6 +1073,11 @@ cdef class SparseOctreeContainer(OctreeContainer):
     def __dealloc__(self):
         # This gets called BEFORE the superclass deallocation.  But, both get
         # called.
+        cdef OctKey *ikey
+        for i in range(self.num_root):
+            ikey = &self.root_nodes[i]
+            tdelete(<void *>ikey, &self.tree_root, root_node_compare)
```

Comment:
I'm struggling to find documentation about this. Can you show me where `tdelete` comes from ?

Reply:
It shows up in `man tdelete` and is part of `search.h` https://pubs.opengroup.org/onlinepubs/009695399/functions/tdelete.html

Reply:
thanks ! I don't feel competent to review this from the technical angle, but I *can* check that this resolves the issue. Will report back as soon as I'm able to.

Reply:
Soooo, actually, it seems to be shadowed by our own version of [tdelete](https://github.com/cphyc/yt/blob/37ad7b5544b5a4a62ffd29ee57c6c81c86ccee2d/yt/utilities/lib/tsearch.c#L75).

Reply:
Oh good!  I forgot we had this.